src/metax_api/api/oaipmh/base/metax_oai_server.py

```diff
@@ -202,16 +210,16 @@ def _get_oai_datacite_metadata(self, record):
         }
         return meta
 
-    def _get_metadata_for_record(self, record, metadata_prefix):
+    def _get_metadata_for_record(self, record, json, type, metadata_prefix):
         meta = {}
 
         # strip sensitive fields from research_dataset. note: the modified research_dataset
         # is placed back into the record's research_dataset -field. meaning, an accidental call
         # of record.save() would overwrite the original data
-        record.research_dataset = CRS.strip_catalog_record(record.research_dataset)
+        record.research_dataset = CRS.strip_catalog_record(json)
```

Comment:
Should be:
json = record.research_dataset = CRS.strip_catalog_record(json)
since later in _get_oai_dc_metadata() the values are taken from variable 'json', not from 'record'. The above long comment is then also made unnecessary

Reply:
json can be both record.research_data and record.catalog_json depending on whether we are getting metadata for CatalogRecord of DataCatalog record instance.

Reply:
It's a bit nasty, I know, but I need both record and record.x and I didn't want to add extra dataset/datacatalog comparisons.

Reply:
But the cleaned "json" value is placed into record.research_dataset, and that value is not used in _get_oai_dc_metadata() ? Below on line 222 self._get_oai_dc_metadata(record, json, type) the var json is not cleaned, and record.reserach_dataset is ?

Reply:
aaaaahh..Not I see.